crates/iceberg-catalog/migrations/20240909120857_configurable_drops.sql

```diff
@@ -0,0 +1,9 @@
+create type tabular_delete_mode as enum ('soft', 'hard');
+
+alter table warehouse
+    add column tabular_expiration_seconds bigint;
+alter table warehouse
+    add column tabular_delete_mode tabular_delete_mode not null default 'soft';
```

Comment:
I am not sure if we should declare a default here. Wouldn't it be better to force us to set this explicitly in the code? Not that what the code does and what we define here diverges because someone forgot to specify?

Reply:
mostly for backwards compat, i'll add & remove the default in the same transaction then